### PR TITLE
Feat/img integration

### PR DESCRIPTION
**Pull request by IsaacGemal**
> [!IMPORTANT]
> Adds an image processing endpoint to analyze food content using Anthropic API and store results with Prisma.
> 
>   - **Behavior**:
>     - New `/upload` endpoint in `index.ts` to process base64 images, analyze for food content, and save results.
>     - Uses `analyzeImageWithAnthropic` from `anthropicService.ts` to analyze images.
>     - Saves analyzed data using `saveFoodData` from `foodService.ts`.
>   - **Services**:
>     - `anthropicService.ts`: Adds `analyzeImageWithAnthropic()` to analyze images using Anthropic API.
>     - `foodService.ts`: Adds `saveFoodData()` to store image and food data in the database.
>     - `testFoodService.ts`: Provides a test function `testFoodService()` to validate saving and cleanup of food data.
>   - **Misc**:
>     - Adds error handling for invalid image formats in `anthropicService.ts`.
>     - Logs detailed request and processing information in `index.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=fractal-bootcamp%2Fbite-tracker&utm_source=github&utm_medium=referral)<sup> for 566029f239a87b57bab10afc69dd12a59e1501e5. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: backend/index.ts [hunk 1]
```diff
@@ -23,79 +25,43 @@ app.use((req, res, next) => {
...
   async (req: express.Request, res: express.Response): Promise<void> => {
     try {
       console.log("Received upload request");
 
-      if (!req.body.image) {
-        console.log("No image data in request");
-        res.status(400).json({ error: "No image data received." });
-        return;
-      }
+      const { image, userId } = req.body;
```
Comment (jwaldor) [inline backend/index.ts]: we don't want to get the user id from the body (as we discussed in slack)

File: backend/services/foodService.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    if (!foodData || !foodData.foodItems || foodData.foodItems.length === 0) {
+      console.log("No food items detected, skipping database save");
+      return null;
+    }
+
+    // First, ensure user exists (create if not)
+    const user = await prisma.user.upsert({
+      where: { clerkId: userId },
+      update: {},
+      create: {
```
Comment (jwaldor) [inline backend/services/foodService.ts]: I don't think we need this. This is handled by the webhook handler which creates a user after a new user is created in Clerk. Of course, it's possible for the webhook handler to fail, but I don't think the foodService should be responsible for handling that specific case explicitly.

File: backend/services/foodService.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    // Create the image with required fields
+    const image = await prisma.image.create({
+      data: {
+        userId: user.id,
+        imageUrl: "https://example.com/image.jpg", // for the future we can store imageData in the cloud
+        foodItems: {
+          create:
+            foodData.foodItems?.map((item: any) => ({
+              name: item.name,
```
Comment (jwaldor) [inline backend/services/foodService.ts:37]: We shouldn't have a ? next to foodItems here. The case where foodItems is undefined should be handled closer to the AI service.

Comment (IsaacGemal) [reply backend/services/foodService.ts:37]: fixed

**Pull request merged**